src/typed-method-types/workflows/triggers/event.ts

```diff
@@ -1,23 +1,109 @@
 import { BaseTrigger, RequiredInputs, TriggerTypes } from "./mod.ts";
+import { FilterType } from "./trigger-filter.ts";
+import { ObjectValueUnion } from "../../../type-helpers.ts";
+import { TriggerEventTypes } from "./trigger-event-types.ts";
 
-type BaseEvent = {
+// TODO: Move to type util file coming in future PR
+// Utility type for the array types which requires minumum one subtype in it.
+type PopulatedArray<T> = [T, ...T[]];
+
+type MessageMetadataTypes = ObjectValueUnion<
+  Pick<
+    typeof TriggerEventTypes,
+    "MessageMetadataPosted" /* | "MessageMetadataAdded" | "MessageMetadataDeleted" */
+  >
+>;
+
+type MessagePostedEventType = ObjectValueUnion<
+  Pick<typeof TriggerEventTypes, "MessagePosted">
+>;
+
+type ChannelTypes = ObjectValueUnion<
```

Comment:
Just FYI, there's been a lot of discussion recently about reworking things so that you can subscribe to a channel-level event (e.g. message_posted) at the workspace level (e.g. "Trip on any message posted event in all channels on this workspace"). So this breakdown between workspace and channel level events may not always be there (or at least, that may change). 

Nothing's concrete right now, but just wanted to give a heads up.